python/ray/autoscaler/_private/aws/node_provider.py

```diff
@@ -365,9 +365,17 @@ def _create_node(self, node_config, tags, count):
                 conf.update({
                     "MinCount": 1,
                     "MaxCount": count,
-                    "SubnetId": subnet_id,
+                    "NetworkInterfaces": [{
+                        "AssociatePublicIpAddress": True,
+                        "DeviceIndex": 0,
+                        "SubnetId": subnet_id,
+                    }],
                     "TagSpecifications": tag_specs
                 })
+                maybe_security_group_ids = conf.pop("SecurityGroupIds", None)
+                if maybe_security_group_ids is not None:
+                    conf["NetworkInterface"]["Groups"] = maybe_security_group_ids
```

Comment:
Looks like test_cli actually did something helpful!
https://buildkite.com/ray-project/ray-builders-pr/builds/8258#92decca1-fb6c-4e5b-ac23-11e4b14b239d/980-1296

"NetworkInterface" -> "NetworkInterfaces"

Reply:
Clearly I'm just testing the tests ;)